src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResource.kt

```diff
@@ -185,6 +185,67 @@ class ActivitiesResource(private val activitiesService: ActivitiesService) {
   fun createOffenderProgramProfile(
     @Schema(description = "Course activity id", required = true) @PathVariable courseActivityId: Long,
     @RequestBody @Valid createRequest: CreateOffenderProgramProfileRequest
-  ): CreateOffenderProgramProfileResponse =
+  ): OffenderProgramProfileResponse =
     activitiesService.createOffenderProgramProfile(courseActivityId, createRequest)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_ACTIVITIES')")
+  @PutMapping("/activities/{courseActivityId}/booking-id/{bookingId}/end")
+  @Operation(
+    summary = "Ends a prisoner's allocation to an activity",
+    description = "Ends a prisoner's allocation to an activity. Requires role NOMIS_ACTIVITIES",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = EndOffenderProgramProfileRequest::class)
+        )
+      ]
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Success",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = """One or more of the following is true:<ul>
```

Comment:
This feels like the kind of documentation that will go stale. But I'm not sure what a better option would be - it's not like we can point at the tests as the definitive list of things that can go wrong, this is a Swaffer doc after all.

Reply:
yes, agree but not sure what is better. This was something Dom mentioned